### PR TITLE
Marcin/extract

### PR DESCRIPTION
**Pull request by marcin-zadrozny**
Added functions to extract.py to find Arxiv papers and extract their .tex files.

### CODE REVIEW

**Review by mknorps — CHANGES_REQUESTED** [review]
Works well at the current stage :rocket: 

I would like to ask you though to add metadata to the data you are downloading from ArXiv.
Each folder should contain a json file called `metadata.json` containing all available metadata given by ArXiv, in particular:
- author's name
- title of the publication
- date of the publication
- domain
- tags

File: rag_experiment/extract.py
Comment (marcin-zadrozny) [inline rag_experiment/extract.py:1]: Not sure if I should include all the other metadata (entry_id, update date etc.), added the requested ones for now.

Comment (mknorps) [reply rag_experiment/extract.py:1]: `entry_id` may be useful for as as a unique identifier of an ArXiv article

File: rag_experiment/extract.py [hunk 1]
```diff
@@ -1,33 +1,103 @@
...
+            'domain': paper.primary_category,
+            'tags': paper.categories
+            }
+        metadata_list.append(metadata)
+    return metadata_list
+        
+        
+# Unpack dowloaded archives, but only .tex files and store them in a separate folder.
+def unpack_archives(source_dir: str, target_dir: str, metadata_list: list):
+    for i, file_name in enumerate(os.listdir(source_dir)):
```
Comment (mknorps) [inline rag_experiment/extract.py]:
```suggestion
    for file_name in os.listdir(source_dir):
```
No need for enumeration. We do not use this `i` anyway..

**Review by mknorps — APPROVED** [review]
LGTM :rocket:

**Pull request merged**